floris/tools/optimization/layout_optimization/layout_optimization_base.py

```diff
@@ -3,15 +3,17 @@
 import numpy as np
 from shapely.geometry import LineString, Polygon
 
+from floris.tools import TimeSeries
 from floris.tools.optimization.yaw_optimization.yaw_optimizer_geometric import (
     YawOptimizationGeometric,
 )
+from floris.tools.wind_data import WindDataBase
 
 from ....logging_manager import LoggingManager
 
 
 class LayoutOptimization(LoggingManager):
-    def __init__(self, fi, boundaries, min_dist=None, freq=None, enable_geometric_yaw=False):
```

Comment:
Does this mean that, if the user would like to provide a weighting to different wind speed bins; they must provide a wind_data object (whereas if they want to use uniform weighting, just supplying fi will suffice)?

Could we instead leave the freq option, which will be ignored if wind_data is None, but which otherwise will be used in the creation of the TimeSeries object in LayoutOptimization.__init__()?

Same goes for the other layout optimization classes, obviously

Reply:
@misi9170 let's chat about this one tomorrow,

Reply:
Ok good point @misi9170 , as discussed changing wind_data to required input to avoid supplying a use case no one is likely to want but could get accidentally